### ISSUE TITLE
Add support for forward mode automatic differentiation

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/pytorch-lightning#12422]
### Motivation

From the [Gradients without Backpropagation](https://arxiv.org/pdf/2202.08587.pdf) (code is coming) and its associated [PyTorch FuncTorch](https://pytorch.org/tutorials/intermediate/forward_ad_usage.html), it seems significant optimization could be made without actually performing the backward pass.

<img width="583" alt="Screenshot 2022-03-23 at 11 59 48" src="https://user-images.githubusercontent.com/12861981/159694303-09b938d5-3933-414e-b065-dc728753423c.png">

Suggestion: This could be added as a new loop as the optimization has backward isn't present.  

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.
